compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1981,7 +1981,7 @@ fn add_upstream_rust_crates<'a, B: ArchiveBuilder<'a>>(
         let filestem = cratepath.file_stem().unwrap().to_str().unwrap();
         cmd.link_rust_dylib(
             Symbol::intern(&unlib(&sess.target, filestem)),
-            parent.unwrap_or(Path::new("")),
+            parent.unwrap_or_else(|| Path::new("")),
```

Comment:
I think the `new` here is probably complex. 

A bit off-topic, but `Path` does not have a `Default`. Is there any reason for this b/c I see `Path::new("")` as a good default.

Reply:
`Path::new` is trivial

Reply:
Repairing it and moving the other question to its own issue.